sql/core/src/main/scala/org/apache/spark/sql/SQLConf.scala

```diff
@@ -319,6 +319,14 @@ private[spark] object SQLConf {
     doc = "When true, some predicates will be pushed down into the Hive metastore so that " +
           "unmatching partitions can be eliminated earlier.")
 
+  val HIVE_NON_NATIVE_VIEW = booleanConf("spark.sql.hive.nonNativeView",
+    defaultValue = Some(false),
+    doc = "When true, CREATE VIEW will be handled by Spark SQL instead of Hive native commands.  " +
+          "Note that this function is experimental and should ony be used when you are using " +
+          "non-hive-compatible tables written by Spark SQL.  The SQL string used to create " +
+          "view should be fully qualified, i.e. use `tbl1`.`col1` instead of `*` whenever " +
+          "possible, or you may get wrong result.")
```

Comment:
how about `spark.sql.canonicalizeView`? Also, can we make it not a public conf? Hopefully, we will have the right mechanism to canonicalize view definitions later.